src/Views/IframeView.php

```diff
@@ -1,88 +1,289 @@
 	 *
 	 * @since 2.7.0
 	 * @var string
 	 */
-	protected $title;
+	protected $url;
 
 	/**
-	 * Document page body.
-	 *
-	 * This will be use to setup content of body tag.
+	 * Flag to check whether show modal or iframe on page.
 	 *
 	 * @since 2.7.0
-	 * @var string
+	 * @var bool
 	 */
-	protected $body;
+	protected $modal = false;
 
 	/**
-	 * Body classes.
+	 * Flag to check whether on not auto scroll page to iframe.
 	 *
-	 * This will be use to setup body tag classes.
+	 * @since 2.7.0
+	 * @var int
+	 */
+	protected $autoScroll = 0;
+
+	/**
+	 * Iframe minimum height.
 	 *
 	 * @since 2.7.0
-	 * @var array
+	 * @var bool
+	 */
+	protected $minHeight = null;
+
+	/**
+	 * Unique identifier.
+	 *
+	 * @var string|null
+	 */
+	protected $uniqueId = null;
+
+	/**
+	 * Button title.
+	 *
+	 * @var string|null
+	 */
+	protected $buttonTitle = null;
+
+	/**
+	 * Button color.
+	 *
+	 * @var string|null
+	 */
+	protected $buttonColor = null;
+
+	/**
+	 * Form template.
+	 *
+	 * @var Template
+	 */
+	protected $template = null;
+
+	/**
+	 * Form id.
+	 *
+	 * @var int
+	 */
+	protected $formId = 0;
+
+	/**
+	 * IframeView Constructor
+	 *
+	 * @param Template $template
+	 */
+	public function __construct( $template = null ) {
+		$this->uniqueId    = uniqid( 'give-' );
+		$this->buttonTitle = __( 'Click to donate', 'give' );
+	}
+
+	/**
+	 * Set iframe URL.
+	 *
+	 * @param string $url
+	 *
+	 * @return $this
+	 */
+	public function setURL( $url = null ) {
+		$this->url = add_query_arg(
+			array_merge( [ 'giveDonationFormInIframe' => 1 ] ),
+			$url
+		);
+
+		return $this;
+	}
+
+	/**
+	 * Set whether or not show modal.
+	 *
+	 * @param bool $set
+	 *
+	 * @return $this
 	 */
-	protected $bodyClasses = [ 'give-form-templates' ];
+	public function showInModal( $set ) {
+		$this->modal = $set;
+
+		return $this;
+	}
 
 	/**
-	 * Set document page title.
+	 * Button title.
 	 *
 	 * @param string $title
 	 *
-	 * @return IframeView $this
+	 * @return $this
 	 */
-	public function setTitle( $title ) {
-		$this->title = $title;
+	public function setButtonTitle( $title ) {
+		$this->buttonTitle = $title;
 
 		return $this;
 	}
 
 	/**
-	 * Set document page title.
+	 * Button color.
 	 *
-	 * @param string $body
+	 * @param string $color
 	 *
-	 * @return IframeView $this
+	 * @return $this
 	 */
-	public function setBody( $body ) {
-		$this->body = $body;
+	public function setButtonColor( $color ) {
+		$this->buttonColor = $color;
 
 		return $this;
 	}
 
 	/**
-	 * Set document page title.
+	 * Form id.
 	 *
-	 * @param array $classes
+	 * @param int $id
 	 *
-	 * @return IframeView $this
+	 * @return $this
 	 */
-	public function setBodyClasses( $classes ) {
-		$this->bodyClasses = array_merge( $this->bodyClasses, (array) $classes );
+	public function setFormId( $id ) {
+		$this->formId = $id;
 
 		return $this;
 	}
 
+	/**
+	 * Get iframe HTML.
+	 *
+	 * @return string
+	 */
+	private function getIframeHTML() {
+		ob_start();
```

Comment:
Need comments around use of object buffering, as it is not clear at a glance what/how this function is doing exactly

Reply:
@bradp I give the title to function, so the developer can understand that we are printing HTML. 

![image](https://user-images.githubusercontent.com/1784821/81197172-9646fb00-8fdd-11ea-8084-39bcf50d7683.png)

I think there is no need to add a comment.

Let me know if you disagree.